homeassistant/components/ecovacs/quality_scale.yaml

```diff
@@ -0,0 +1,95 @@
+rules:
+  # Bronze
+  config-flow: done
+  test-before-configure: done
+  unique-config-entry: done
+  config-flow-test-coverage: done
+  runtime-data: done
+  test-before-setup:
+    status: todo
+    comment: Legacy code will not raise on setup currently
+  appropriate-polling:
+    status: todo
+    comment: |
+      @mib1185 Please check legacy code.
+      deebot-client pulls only once at beginning and afterwards is pushed based
+  entity-unique-id: done
+  has-entity-name: done
+  entity-event-setup: done
+  dependency-transparency:
+    status: todo
+    comment: Currently unsure if all dependencies need to validated or only direct ones.
+  action-setup:
+    status: done
+    comment: "`raw_get_positions` is a entity service"
+  common-modules: done
+  docs-high-level-description: todo
+  docs-installation-instructions: todo
+  docs-removal-instructions: todo
+  docs-actions: todo
+  brands: done
+
+  # Silver
+  config-entry-unloading: done
+  log-when-unavailable: todo
+  entity-unavailable: done
+  action-exceptions: todo
+  reauthentication-flow: todo
+  parallel-updates:
+    status: todo
+    comment: |
+      @mib1185 Please check legacy code.
+      deebot-client uses internally semaphores to prevent to many parallel requests
+  test-coverage: todo
+  integration-owner: done
+  docs-installation-parameters: todo
+  docs-configuration-parameters: todo
+
+  # Gold
+  entity-translations:
+    status: todo
+    comment: |
+      @mib1185 Legacy entities are not translated
+  entity-device-class: done
+  devices: done
+  entity-category: done
+  entity-disabled-by-default: done
+  discovery:
+    status: exempt
+    comment: Not supported as we don't talk directly to the devices
```

Comment:
We can still discover them.
Did you check:
- mDNS messages
- MAC registered as OUI
- Registers on the network with a default hostname

Reply:
Ecovacs has a config entry per cloud account and not per device. It makes no sense to even check for discovery if we can't interact with the device directly but only over the cloud service. The cloud API is rest and the bot only speaks MQTT with no way to change the broker.

I'm sorry, but I don't see any advantage to implementing discovery if you need the username and password for the cloud service. Also if you have already setup your account the bot will be added automatically on reboot (maybe I will add automatic discovery in the future)